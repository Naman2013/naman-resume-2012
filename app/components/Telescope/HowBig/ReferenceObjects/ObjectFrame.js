import React from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';

const ObjectFrame = ({
  svgURL,
  onLoadCallback,
}) => (
  <g>
    <image
      className="object-frame-image"
      xlinkHref={svgURL}
      width="100%"
      height="100%"
      onLoad={() => { onLoadCallback(); }}
    />
  </g>
);

ObjectFrame.propTypes = {
  svgURL: PropTypes.oneOf([PropTypes.string, PropTypes.shape({})]).isRequired,
  onLoadCallback: PropTypes.func,
};

ObjectFrame.defaultProps = {
  onLoadCallback: noop,
};

export default ObjectFrame;
