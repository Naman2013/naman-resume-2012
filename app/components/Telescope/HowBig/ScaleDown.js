import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ObjectFrame from './ReferenceObjects/ObjectFrame';
import FadeSVG from 'components/common/Fade/FadeSVG';
import domains from './domains';

class ScaleDown extends Component {
  static propTypes = {
    referenceObject: PropTypes.oneOf([
      'SOLAR_SYSTEM',
      'STAR',
      'MILKY_WAY',
      'DEEP_SPACE',
    ]).isRequired,
    targetObjectURL: PropTypes.string.isRequired,
    targetObjectScale: PropTypes.number.isRequired,
    onComplete: PropTypes.func.isRequired,
  };

  state = {
    targetCurrentScale: 1,
    targetObjectLoaded: false,
    referenceObjectLoaded: false,
    referenceOpacity: 1,
  };

  render() {
    const {
      referenceObject,
      targetObjectURL,
      dimension,
    } = this.props;

    const {
      targetCurrentScale,
      referenceOpacity,
    } = this.state;

    return (
      <g style={{ transformOrigin: 'center', transform: 'scale(0.85)' }}>
        <g style={{
          transform: 'translate(0, 0) scale(1)',
          opacity: referenceOpacity,
        }}
        >
          {domains.enumValueOf(referenceObject).render()}
        </g>

        <g style={{
            transform: `translate(0, 0) scale(${targetCurrentScale})`,
          }}
        >
          <ObjectFrame />
        </g>
      </g>
    );
  }
}

export default ScaleDown;
