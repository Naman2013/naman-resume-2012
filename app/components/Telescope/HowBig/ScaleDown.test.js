import React from 'react';
import { shallow } from 'enzyme';
import ScaleDown from './ScaleDown';
import ObjectFrame from './ReferenceObjects/ObjectFrame';

describe('ScaleDown', () => {
  const initialProps = {
    referenceObject: 'SOLAR_SYSTEM',
    targetObjectURL: 'https://vega.slooh.com/icons/community/human_spirit.svg',
    targetObjectScale: 0.34,
    dimension: 500,
    onComplete: jest.fn(),
  };
  const scaleDown = shallow(<ScaleDown {...initialProps} />);

  describe('initialization', () => {
    it('should render correctly', () => {
      expect(scaleDown).toMatchSnapshot();
    });

    it('should initialize state with `targetInitialSize`', () => {
      expect(scaleDown.state().targetCurrentScale).toEqual(1);
    });

    it('should initialize state with `referenceOpacity` to 1', () => {
      expect(scaleDown.state().referenceOpacity).toEqual(1);
    });

    it('should initialize state with `referenceObjectLoaded` and `targetObjectLoaded`', () => {
      expect(scaleDown.state().referenceObjectLoaded).toEqual(false);
      expect(scaleDown.state().targetObjectLoaded).toEqual(false);
    });
  });

  describe('when provided a target domain/reference', () => {
    it('will render two `ObjectFrame`s', () => {
      expect(scaleDown.find(ObjectFrame).length).toEqual(2);
    });
  });

  describe('when loading reference and target svg elements', () => {
    beforeEach(() => {
      scaleDown.setState({ targetObjectLoaded: false, referenceObjectLoaded: false });
    });

    it('should set `targetObjectLoaded` to true', () => {
      scaleDown.instance().handleTargetObjectLoaded();
      expect(scaleDown.state().targetObjectLoaded).toEqual(true);
    });

    it('should set `referenceObjectLoaded` to true', () => {
      scaleDown.instance().handleReferenceObjectLoaded();
      expect(scaleDown.state().referenceObjectLoaded).toEqual(true);
    });
  });

  describe('when it completes animating', () => {
    xit('should call the onComplete function', () => {
      expect(initialProps.onComplete).toBeCalled();
    });
  });
});
