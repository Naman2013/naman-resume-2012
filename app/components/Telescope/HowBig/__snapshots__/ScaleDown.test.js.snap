// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScaleDown initialization should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScaleDown
    dimension={500}
    onComplete={[MockFunction]}
    referenceObject="SOLAR_SYSTEM"
    targetObjectScale={0.34}
    targetObjectURL="https://vega.slooh.com/icons/community/human_spirit.svg"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <g
          style={
            Object {
              "opacity": 1,
              "transform": "translate(0, 0) scale(1)",
            }
          }
        >
          <ObjectFrame
            onLoadCallback={[Function]}
            svgURL={
              Object {
                "process": [Function],
              }
            }
          />
        </g>,
        <g
          style={
            Object {
              "transform": "translate(0, 0) scale(1)",
            }
          }
        >
          <ObjectFrame
            onLoadCallback={[Function]}
          />
        </g>,
      ],
      "style": Object {
        "transform": "scale(0.85)",
        "transformOrigin": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ObjectFrame
            onLoadCallback={[Function]}
            svgURL={
              Object {
                "process": [Function],
              }
            }
          />,
          "style": Object {
            "opacity": 1,
            "transform": "translate(0, 0) scale(1)",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onLoadCallback": [Function],
            "svgURL": Object {
              "process": [Function],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "g",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ObjectFrame
            onLoadCallback={[Function]}
          />,
          "style": Object {
            "transform": "translate(0, 0) scale(1)",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onLoadCallback": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "g",
      },
    ],
    "type": "g",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <g
            style={
              Object {
                "opacity": 1,
                "transform": "translate(0, 0) scale(1)",
              }
            }
          >
            <ObjectFrame
              onLoadCallback={[Function]}
              svgURL={
                Object {
                  "process": [Function],
                }
              }
            />
          </g>,
          <g
            style={
              Object {
                "transform": "translate(0, 0) scale(1)",
              }
            }
          >
            <ObjectFrame
              onLoadCallback={[Function]}
            />
          </g>,
        ],
        "style": Object {
          "transform": "scale(0.85)",
          "transformOrigin": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ObjectFrame
              onLoadCallback={[Function]}
              svgURL={
                Object {
                  "process": [Function],
                }
              }
            />,
            "style": Object {
              "opacity": 1,
              "transform": "translate(0, 0) scale(1)",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onLoadCallback": [Function],
              "svgURL": Object {
                "process": [Function],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "g",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ObjectFrame
              onLoadCallback={[Function]}
            />,
            "style": Object {
              "transform": "translate(0, 0) scale(1)",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onLoadCallback": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "g",
        },
      ],
      "type": "g",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
